security/pkg/nodeagent/vm/nodeagent.go

```diff
@@ -59,26 +64,70 @@ func (na *nodeAgentInternal) Start() error {
 	na.identity = identity
 	var success bool
 	for {
-		privateKey, req, reqErr := na.createRequest()
-		if reqErr != nil {
-			return reqErr
+		var csrResponseFromOldAPI *pb.CsrResponse
+		var keyPEM []byte
+		var csrPEM []byte
+		var certChainPEM []string
+		var csrError error
+		if na.config.CAClientConfig.IsCitadel {
+			privateKey, req, reqErr := na.createRequest()
+			if reqErr != nil {
+				return reqErr
+			}
+			csrResponseFromOldAPI, csrError = na.caProtocol.SendCSR(req)
+			keyPEM = privateKey
+		} else {
+			options := pkiutil.CertOptions{
+				Host:       identity,
+				Org:        na.config.CAClientConfig.Org,
+				RSAKeySize: na.config.CAClientConfig.RSAKeySize,
+				IsDualUse:  na.config.DualUse,
+			}
+
+			csrPEM, keyPEM, err = pkiutil.GenCSR(options)
+			if err != nil {
+				log.Errorf("Failed to generated key cert for %q: %v", identity, err)
+			}
+			jwt, err := na.pc.GetAgentCredential()
+			if err != nil {
+				log.Errorf("Error getting JWT from metadata server: %v", err)
+			}
+			stsClient := stsclient.NewPlugin()
+			// TODO(pitlv2109): Periodically update the JWT and OAuth2 token.
+			// TODO(pitlv2109): Extract the project_id field from the JWT.
+			token, _, _ := stsClient.ExchangeToken(context.Background(), "csm-fsa@mesh-expansion-235023.iam.gserviceaccount.com", string(jwt[:]))
+			certChainPEM, csrError = na.caClient.CSRSign(context.Background(), csrPEM, token, int64(na.config.CAClientConfig.RequestedCertTTL.Minutes()))
 		}
 
 		log.Infof("Sending CSR (retrial #%d) ...", retries)
 
-		resp, err := na.caProtocol.SendCSR(req)
-		if err == nil && resp != nil && resp.IsApproved {
-			waitTime, ttlErr := na.certUtil.GetWaitTime(resp.SignedCert, time.Now())
+		if na.isCsrRequestApproved(csrResponseFromOldAPI, csrError) {
+			var waitTime time.Duration
+			var ttlErr error
+			if na.config.CAClientConfig.IsCitadel {
+				waitTime, ttlErr = na.certUtil.GetWaitTime(csrResponseFromOldAPI.SignedCert, time.Now())
```

Comment:
Add a pointer of cert and using one if else to decide if that should be obtained from OldResponse or cerchainPEM[0].